test/integration/node-specific/bson-options/useBigInt64.test.ts

```diff
@@ -127,25 +127,119 @@ describe('useBigInt64 option', function () {
     });
   });
 
-  describe('when set to true and promoteLongs is set to false', function () {
-    it('throws a MongoAPIError', async function () {
-      expect(() => {
-        configuration.newClient(configuration.writeConcernMax(), {
-          useBigInt64: true,
-          promoteLongs: false
-        });
-      }).to.throw(MongoAPIError, /Must request either bigint or Long for int64 deserialization/);
+  describe('when useBigInt64=true and promoteLongs=false', function () {
+    let client: MongoClient;
+
+    afterEach(async function () {
+      if (client) {
+        await client.close();
+      }
+    });
+
+    describe('when set at client level', function () {
+      it('throws a MongoAPIError', async function () {
+        expect(() => {
+          client = configuration.newClient(configuration.writeConcernMax(), {
+            useBigInt64: true,
+            promoteLongs: false
+          });
+        }).to.throw(MongoAPIError, /Must request either bigint or Long for int64 deserialization/);
+      });
+    });
+
+    describe('when set at DB level', function () {
+      it('throws a BSONError', async function () {
+        client = configuration.newClient(configuration.writeConcernMax());
+        await client.connect();
```

Comment:
```suggestion
```

(optional - applies to all tests here that connect) we can omit the connecting if we want, since the driver supports auto-connection now